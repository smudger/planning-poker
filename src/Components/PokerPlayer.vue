<script setup>
import { computed } from "vue";
import { minidenticon } from "minidenticons";
import { CheckIcon } from "@heroicons/vue/24/solid";

const props = defineProps({
  name: { type: String, default: undefined },
  estimate: { type: [String, Number], default: undefined },
  showEstimate: { type: Boolean, required: true },
});

const avatar = computed(
  () =>
    "data:image/svg+xml;utf8," + encodeURIComponent(minidenticon(props.name)),
);
</script>

<template>
  <div
    :class="[
      name ? 'bg-gray-800 border-gray-800' : 'border-dashed border-gray-300',
    ]"
    class="rounded-2xl border-2 justify-between flex w-72 items-center px-6 py-4 space-x-2"
  >
    <h3
      class="text-base font-semibold leading-7 tracking-tight text-white whitespace-nowrap truncate"
    >
      {{ name }}
    </h3>
    <div
      v-if="showEstimate && name"
      :class="[estimate ? 'bg-indigo-600' : '']"
      class="h-16 w-16 shrink-0 flex items-center justify-center rounded-full"
    >
      <span
        class="text-white font-mono font-semibold leading-7 tracking-tight text-xl"
      >
        {{ estimate }}
      </span>
    </div>
    <div v-else-if="name" class="relative">
      <img
        :class="[estimate ? 'ring-4 ring-indigo-600' : '']"
        :src="avatar"
        alt="avatar"
        class="h-16 w-16 shrink-0 rounded-full bg-white"
      />
      <span
        v-if="estimate"
        class="absolute right-0 top-0 block h-4 w-4 rounded-full bg-indigo-600 ring-2 ring-white text-white"
      >
        <CheckIcon />
      </span>
    </div>
    <div v-else class="h-16 w-16 shrink-0 rounded-full bg-gray-300" />
  </div>
</template>
