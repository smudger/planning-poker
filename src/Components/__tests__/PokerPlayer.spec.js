import { afterEach, describe, expect, it, vi } from "vitest";
import { shallowMount } from "@vue/test-utils";
import PokerPlayer from "@/Components/PokerPlayer.vue";

vi.mock("minidenticons", () => {
  return { minidenticon: () => "image-url" };
});

describe("PokerPlayer", () => {
  afterEach(() => {
    vi.clearAllMocks();
  });

  describe("when the player has a name", () => {
    it("displays the player's name", () => {
      const wrapper = shallowMount(PokerPlayer, {
        props: {
          name: "Constance Wu",
          estimate: 3,
          showEstimate: false,
        },
      });

      expect(wrapper.text()).toContain("Constance Wu");
    });

    it("displays an avatar based on the player's name", () => {
      const wrapper = shallowMount(PokerPlayer, {
        props: {
          name: "Constance Wu",
          estimate: 3,
          showEstimate: false,
        },
      });

      expect(wrapper.find("img").exists()).toBe(true);
      expect(wrapper.find("img").attributes("src")).toBe(
        "data:image/svg+xml;utf8,image-url",
      );
    });
  });

  describe("when the player does not have a name", () => {
    it("does not display any text", () => {
      const wrapper = shallowMount(PokerPlayer, {
        props: {
          name: undefined,
          estimate: undefined,
          showEstimate: false,
        },
      });

      expect(wrapper.text()).toBe("");
    });

    it("does not display an avatar", () => {
      const wrapper = shallowMount(PokerPlayer, {
        props: {
          name: undefined,
          estimate: undefined,
          showEstimate: false,
        },
      });

      expect(wrapper.find("img").exists()).toBe(false);
    });
  });

  it("displays the player's estimate when the show estimate prop is true", () => {
    const wrapper = shallowMount(PokerPlayer, {
      props: {
        name: "Constance Wu",
        estimate: 3,
        showEstimate: true,
      },
    });

    expect(wrapper.text()).toContain("3");
  });

  it("does not display the player's estimate when the show estimate prop is false", () => {
    const wrapper = shallowMount(PokerPlayer, {
      props: {
        name: "Constance Wu",
        estimate: 3,
        showEstimate: false,
      },
    });

    expect(wrapper.text()).not.toContain("3");
  });
});
