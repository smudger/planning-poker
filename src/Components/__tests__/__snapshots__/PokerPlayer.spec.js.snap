// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokerPlayer > renders correctly when show estimate is false 1`] = `
<div
  class="bg-gray-800 border-gray-800 rounded-2xl border-2 justify-between flex sm:flex-col-reverse w-80 sm:w-64 items-center px-6 py-4 space-x-2 sm:space-x-0"
>
  <div
    class="sm:mt-4"
  >
    <h3
      class="text-base font-semibold leading-7 tracking-tight text-white whitespace-nowrap truncate"
    >
      Constance Wu
    </h3>
  </div>
  <div>
    <div
      class="relative"
    >
      <img
        alt="avatar"
        class="ring-4 ring-indigo-600 h-16 w-16 sm:h-32 sm:w-32 shrink-0 rounded-full bg-white"
        src="data:image/svg+xml;utf8,image-url"
      />
      <span
        class="absolute right-0 top-0 sm:right-1 sm:top-1 block h-4 sm:h-6 w-4 sm:w-6 rounded-full bg-indigo-600 ring-2 ring-white text-white"
      >
        <render-stub />
      </span>
    </div>
  </div>
</div>
`;

exports[`PokerPlayer > renders correctly when show estimate is true 1`] = `
<div
  class="bg-gray-800 border-gray-800 rounded-2xl border-2 justify-between flex sm:flex-col-reverse w-80 sm:w-64 items-center px-6 py-4 space-x-2 sm:space-x-0"
>
  <div
    class="sm:mt-4"
  >
    <h3
      class="text-base font-semibold leading-7 tracking-tight text-white whitespace-nowrap truncate"
    >
      Constance Wu
    </h3>
  </div>
  <div>
    <div
      class="bg-indigo-600 h-16 w-16 sm:h-32 sm:w-32 shrink-0 flex items-center justify-center rounded-full"
    >
      <span
        class="text-white font-mono font-semibold leading-7 tracking-tight text-xl sm:text-4xl"
      >
        3
      </span>
    </div>
  </div>
</div>
`;

exports[`PokerPlayer > renders correctly when there is no name 1`] = `
<div
  class="border-dashed border-gray-300 rounded-2xl border-2 justify-between flex sm:flex-col-reverse w-80 sm:w-64 items-center px-6 py-4 space-x-2 sm:space-x-0"
>
  <div
    class="sm:mt-4"
  >
    <div
      class="rounded-2xl w-32 h-3 my-2 bg-gray-300"
    />
  </div>
  <div>
    <div
      class="h-16 w-16 sm:h-32 sm:w-32 shrink-0 rounded-full bg-gray-300"
    />
  </div>
</div>
`;
